test/sharness/t0700-remotepin.sh

```diff
@@ -0,0 +1,177 @@
+  ipfs pin remote ls --service=test_pin_svc --enc=json
+'
+
+# TODO: improve error returned below
+test_expect_failure "unathorized pinning service calls fail" '
+  ipfs pin remote ls --service=test_invalid_key_svc
+'
+
+# TODO: improve error returned below
+test_expect_failure "misconfigured pinning service calls fail (wrong path)" '
+  ipfs pin remote ls --service=test_invalid_url_path_svc
+'
+
+# TODO: improve error returned below (panic when offline mode)
+test_expect_failure "misconfigured pinning service calls fail (dns error)" '
+  ipfs pin remote ls --service=test_invalid_url_dns_svc
+'
+
+test_expect_success "remove pinning service" '
+  ipfs pin remote service rm test_invalid_key_svc &&
+  ipfs pin remote service rm test_invalid_url_path_svc &&
+  ipfs pin remote service rm test_invalid_url_dns_svc
+'
+
+test_expect_success "verify pinning service removal works" '
+  ipfs pin remote service ls | jq --raw-output .Service | grep -L test_invalid_key_svc
+'
+
+# TODO: why this was not pinned instantly? no bitswap is required for inlined CID
+test_expect_success "verify background add works with data inlined in CID" '
+  ipfs pin remote add --background=true --service=test_pin_svc --enc=json --name=inlined_null bafkqaaa &&
+  ipfs pin remote ls --service=test_pin_svc --enc=json --name=inlined_null --status=queued --status=pinning --status=pinned > ls_out &&
+  grep -E "queued|pinning|pinned" ls_out
+'
+
+test_remote_pins() {
+  BASE=$1
+  if [ -n "$BASE" ]; then
+    BASE_ARGS="--cid-base=$BASE"
+  fi
+
+  # time-based CIDs to ensure we test with data that is not on public network
+  test_expect_success "create some hashes using base $BASE" '
+    export HASH_A=$(echo "A @ $(date)" | ipfs add $BASE_ARGS -q --pin=false) &&
+    export HASH_B=$(echo "B @ $(date)" | ipfs add $BASE_ARGS -q --pin=false) &&
+    export HASH_C=$(echo "C @ $(date)" | ipfs add $BASE_ARGS -q --pin=false) &&
+    export HASH_D=$(echo "D @ $(date)" | ipfs add $BASE_ARGS -q --pin=false)
+  '
+
+  # TODO: when running locally, background=false seems to block forever - perhaps due to missing Pin.origins?
+  test_expect_success "'ipfs pin remote add'" '
+    export ID_A=$(ipfs pin remote add --background=false --service=test_pin_svc --enc=json $BASE_ARGS --name=name_a $HASH_A | jq --raw-output .RequestID) &&
+    export ID_B=$(ipfs pin remote add --background=false --service=test_pin_svc --enc=json $BASE_ARGS --name=name_b $HASH_B | jq --raw-output .RequestID) &&
+    export ID_C=$(ipfs pin remote add --background=false --service=test_pin_svc --enc=json $BASE_ARGS --name=name_c $HASH_C | jq --raw-output .RequestID) &&
+    export ID_D=$(ipfs pin remote add --background=true --service=test_pin_svc --enc=json $BASE_ARGS --name=name_d $HASH_D | jq --raw-output .RequestID)
+  '
+
+  test_expect_success "verify background add worked" '
+    ipfs pin remote ls --service=test_pin_svc --enc=json --name=name_d --status=queued --status=pinning --status=pinned > ls_out &&
```

Comment:
I'd avoid anything that is testing external state when it does not impact the execution of `pin remote` command. 
The only time sharness should care about pin status being `pinned` is when `--bacground=false` is passed.

Reply:
I mostly agree with the concept, but testing both here verifies we don't run into regressions where the code paths for `--background=false|true` diverge too much

Reply:
I think if we can make [rb-pinning-service-api](https://github.com/ipfs-shipyard/rb-pinning-service-api) immediately return inlined CIDs (example: `bafkqaaa`  – empty identity hash)  in "pinned" status (because no data needs to be fetched, its inlined in the CID), then we have means of testing `--background=false` on CI reliably.

cc @andrew – is that doable?